### PR TITLE
trigger status notifications for all connectors

### PR DESCRIPTION
**Pull request by lbbrhzn**

### CODE REVIEW

File: custom_components/ocpp/api.py [hunk 1]
```diff
@@ -490,16 +490,18 @@ async def trigger_boot_notification(self):
...
-        )
-        resp = await self.call(req)
-        if resp.status == TriggerMessageStatus.accepted:
-            return True
-        else:
-            _LOGGER.warning("Failed with response: %s", resp.status)
-            return False
+        """Trigger status notifications for all connectors."""
+        return_value = True
+        for id in range(0, 1):
```
Comment (drc38) [inline custom_components/ocpp/api.py:495]: Use `self._metrics[cdet.connectors.value].value` here instead?

**Comment by drc38** [comment]
Perhaps in  the `@on(Action.StatusNotification)` use the extra attributes of the connector status metric to handle statuses and errors for more than 1 connector?

**Comment by lbbrhzn** [comment]
Not sure what you mean. The functionality to store status and error code separately for  main (id=0) and connector (id>0) is already in.

**Comment by drc38** [comment]
> Not sure what you mean. The functionality to store status and error code separately for main (id=0) and connector (id>0) is already in.

Is it for only 1 (ie 0 and 1) connector or will it handle say 0, 1 and 2?

**Comment by lbbrhzn** [comment]
We could support more than one connector, but to display the info in separate sensors we would need to know how many there at setup, or create the additional sensors on the fly, which is messy.

Alternatively we could just keep the two sensors for connector=0 and connector=1 as is, and store additional info for other connectors in extra_attributes.

File: custom_components/ocpp/api.py (hunk 1 above)
Comment (lbbrhzn) [reply custom_components/ocpp/api.py:495]: It does need to be known, so the number of connectors should be retrieved before sending the triggers.

**Comment by drc38** [comment]
> So, let's just keep the two sensors for connector=0 and connector=1 as is, and store additional info for other connectors in extra_attributes.

Agree 👍

**Pull request merged**